File: C#/MinEatingSpeed.cs
public class Solution {
    public int MinEatingSpeed(int[] piles, int h) {
        
        float k = 1;
        double dayCount = 0;
        
        while (true)
        {
            dayCount = 0;
            
            //Console.WriteLine("K value: " + k);
            
            for (int i = 0; i < piles.Length; i++)
            {
                //Console.WriteLine( Math.Ceiling((decimal)(piles[i] / k)) );
                dayCount += (double)Math.Ceiling((decimal)(piles[i] / k));
                
                if (dayCount > h)
                {
                    break;
                }
            }
            
            //Console.WriteLine("Day Count: " + dayCount);
            //Console.Write("\n");
            
            if (dayCount > h)
            {
                k++;
            }
            else
            {
                return ((int)k);
            }
            
        }
        
        return 0;
    }
}

File: Python/minEatingSpeed.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:

        k = 1
        while (True):
            hourCount = 0

            for pile in piles:

                hourCount += math.ceil(pile / k)
                
            if hourCount <= h:
                return k

            k += 1
